File: game.js
const canvas = document.getElementById('gameCanvas');
const ctx = canvas.getContext('2d');

const playerShip = new PlayerShip(10, 10);
const enemyShips = [
    new EnemyShip(5, 5),
    new EnemyShip(15, 15),
    new EnemyShip(20, 20)
];

function handlePlayerInput(event) {
    switch (event.key) {
        case 'ArrowUp':
            playerShip.move('up');
            break;
        case 'ArrowDown':
            playerShip.move('down');
            break;
        case 'ArrowLeft':
            playerShip.move('left');
            break;
        case 'ArrowRight':
            playerShip.move('right');
            break;
        case 'w':
            playerShip.move('up-left');
            break;
        case 'e':
            playerShip.move('up-right');
            break;
        case 's':
            playerShip.move('down-left');
            break;
        case 'd':
            playerShip.move('down-right');
            break;
        case ' ':
            playerShip.shoot();
            break;
    }
}

document.addEventListener('keydown', handlePlayerInput);

function gameLoop() {
    ctx.clearRect(0, 0, canvas.width, canvas.height);

    playerShip.draw(ctx);
    enemyShips.forEach(ship => {
        ship.moveRandomly();
        ship.draw(ctx);
    });

    requestAnimationFrame(gameLoop);
}

gameLoop();

File: old/game.py
from browser import document, html, timer
from playerShip import PlayerShip
from enemyShip import EnemyShip
from main import handlePlayerInput, gameLoop

canvas = document["gameCanvas"]
ctx = canvas.getContext("2d")

player_ship = PlayerShip(10, 10)
enemy_ships = [
    EnemyShip(5, 5),
    EnemyShip(15, 15),
    EnemyShip(20, 20)
]

turn = 0

def handle_keydown(event):
    handlePlayerInput(event)

document.bind("keydown", handle_keydown)

def start_game():
    gameLoop()

timer.set_interval(start_game, 1000 // 60)
